Technical/Performance-Benchmarking/arangodb-clustering-performance-report.md

```diff
@@ -0,0 +1,700 @@
+# ArangoDB Clustering Performance Report
+
+**Summary:**
+
+This report presents the findings from the recent performance testing focused on the ArangoDB clustering capabilities. The tests were designed to evaluate the efficiency of read and write operations within a clustered environment, using JMeter as the tool to generate the workload. The main points of consideration were the processing times for read queries, CPU utilization on Azure nodes, write operation throughput and impact of scaling operations. 
+
+**Test Environment:**
+
+- ArangoDB Cluster vs singe deployment test
+- Cluster Configuration: Initially 3 nodes with 96 vCPUs; scaled to 4 nodes
+- Load Testing Tool: Apache JMeter
+- Deployment: Azure Kubernetes Service (AKS)
+
+**Test Execution:**
+1.	**Cluster Configuration & Metrics:**
+- The ArangoDB cluster was initially set up with 3 Azure nodes, each with 96 vCPUs. Due to performance concerns (maxing CPU), an additional node was added to the cluster.
+- Metrics were gathered from the ArangoDB dashboard, showing CPU usage and other performance-related statistics.
+2.	**Read Queries Performance:**
+- Processing times for read queries were found to be slow, which correlated with high CPU utilization, causing the cluster to max out the CPU resources causing E2E to be extremely slow.
+- The Azure metrics indicated that as the CPU load increased, the read operation latencies also increased, leading to performance bottlenecks.
+3.	**Write Operations Performance:**
+- JMeter tests showed that write operations to ArangoDB were completed quickly, with an ingestion rate of 2500 to 2700 messages per second.
+- All transactions were submitted to the Transaction Monitoring Service (TMS) within a 10-second timeframe, indicating that the writes were not affected by the issues impacting read operations.
+4.	**Database Scaling:**
+- To address the performance issues, the database instances were scaled from 3 to 4.
+- An additional Azure node was provisioned to host the new database instance, aimed at distributing the load and improving read query performance.
+5.	**Network Topology Optimization:**
+- The typologies were changed from 26 to 7  to improve the performance of the Typology Processor, Transaction Aggregatorand Decision Processor.
+- This change was made in response to the observed delays in rule processing.
```

Comment:
Removed this section , because thats true the rules were the same so the reads would have been the same . I just reduced the TPs to try figure out what was causing the horrendous FTPS.